docs/usage/templates.md

```diff
@@ -85,3 +85,16 @@ In the example above, it will only show a text if `isMajor=true` and `hasRelease
 Returns `true` if at least one expression is `true`.
 
 `{{#if (or isPatch isSingleVersion}}Small update, safer to merge and release.{{else}}Check out the changelog for all versions before merging!{{/if}}`
+
+## Environment variables
+
+By default, you can only access a handful of basic environment variables like `HOME` or `PATH`.
+This is for security reasons.
+
+`HOME is {{env.HOME}}`
+
+If you're self-hosting Renovate, you can expose additional variables with the [`customEnvVariables`](https://docs.renovatebot.com/self-hosted-configuration/#customenvvariables) config option.
+
+You can also use the [`exposeAllEnv`](https://docs.renovatebot.com/self-hosted-configuration/#exposeallenv) config option to allow all environment variables in templates.
+Always consider the security implications of using `exposeAllEnv`.
+Secrets and other confidential information stored in environment variables could be leaked by a malicious script.
```

Comment:
@HonkingGoose @rarkins should we add a bigger warning here?

Reply:
We should use a warning admonition to grab the readers attention.

I think the key points to cover are:

- Explain what's the benefit of enabling the dangerous option
- Explain why it can be dangerous
- If there's any way to make it less dangerous, explain how
- Give our recommendations as maintainers
- Warn loudly about dangerous stuff

Reply:
can you make a suggestion?

Reply:
I think exposeAllEnv should have the big warning while this docs section just references it

Reply:
I added the big warning to `docs/usage/self-hosted-configuration.md` and left only a short remark in `docs/usage/templates.md`.